function/colcon_cd.sh

```diff
@@ -0,0 +1,79 @@
+# copied from colcon_cd/function/colcon_cd.sh
+
+colcon_cd() {
+  if [ $# = 0 ]; then
+    # change the working directory to the previously saved path
+    if [ "$_colcon_cd_root" = "" ]; then
+      echo "No previous path saved. Either call 'colcon_cd <pkgname>' from a" \
+        "directory where <pkgname> can be found or 'colcon_cd --set' to" \
+        "directly save the current working directory." 1>&2
+      return 1
+    fi
+    if [ "$_colcon_cd_root" != "$(pwd)" ]; then
+      cd "$_colcon_cd_root"
+    fi
+
+  elif [ $# = 1 ]; then
+    if [ "$1" = "--set" ]; then
+      # store the current working directory for future invocations
+      _colcon_cd_root="$(pwd)"
+      echo "Saved the current working directory for future invocations of" \
+        "'colcon_cd <pkgname>' as well as to return to using 'colcon_cd'." \
+        "Call 'colcon_cd --reset' to unset the saved path."
+      return 0
+    elif [ "$1" = "--reset" ]; then
+      # unset the save path
+      unset _colcon_cd_root
+      return 0
+    fi
+
+    _colcon_cd_cmd="colcon list --packages-select $1 --paths-only"
+
+    if [ "$_colcon_cd_root" != "" ]; then
+      # try to find the given package from the saved path
+      _colcon_cd_pwd="$(pwd)"
+      cd "$_colcon_cd_root"
```

Comment:
Using a `find` command like above, we wouldn't have to change directories, instead just changing the path given to `find`.

Reply:
See the other comment why `find` isn't an option.

The `cd` would be avoidable if the `colcon list` command could be invoked from a different directory. That is not the case though. See colcon/colcon-core#139 for a similar request.